Exercise_2.cpp

```diff
@@ -19,21 +20,45 @@ StackNode* newNode(int data)
 int isEmpty(StackNode* root) 
 { 
     //Your code here 
+    return root?0:1;
+
 } 
   
 void push(StackNode** root, int data) 
 { 
     //Your code here 
+    StackNode* t = new StackNode;

```

Comment:
This implementation will work but not common where you add a new node before the head. Please add comments and time and space complexity